astropy/cosmology/parameter/tests/test_descriptors.py

```diff
@@ -3,15 +3,70 @@
 """Testing :mod:`astropy.cosmology.parameter._descriptor`."""
 
 from types import MappingProxyType
+from typing import ClassVar
 
 import pytest
 
 from astropy.cosmology._utils import all_cls_vars
 from astropy.cosmology.parameter import Parameter
+from astropy.cosmology.parameter._descriptors import ParametersAttribute
+
+
+class Obj:
+    a: ClassVar = 1
+    b: ClassVar = 2
+    c: ClassVar = 3
+    _attr_map: ClassVar = ("a", "b", "c")
+
+    attr = ParametersAttribute(attr_name="_attr_map")
+
+
+class TestParametersAttribute:
+    """Test the descriptor ``ParametersAttribute``."""
+
+    @pytest.fixture
```

Comment:
Why these fixtures? They just make the test super hard to read. 

I'd also suggest to define the `Obj` class inside this test class, maybe as part of `setup_class`, since it is not used elsewhere:
```
@classmethod
def setup_class(self):
    class Obj:
        ...

    self.Obj = Obj
    self.obj = Obj()
```

Reply:
https://docs.pytest.org/en/6.2.x/xunit_setup.html

"Note While these setup/teardown methods are simple and familiar to those coming from a unittest or nose background, you may also consider using pytest’s more powerful [fixture mechanism](https://docs.pytest.org/en/6.2.x/fixture.html#fixture) which leverages the concept of dependency injection, allowing for a more modular and more scalable approach for managing test state"

Reply:
In that case, maybe keep the test class outside and create `obj = Obj()` in the two tests that need it?  Readability counts...  

Though that also implies my suggestion of `setup_class` is not much better -- best to just keep it simple for putative future contributors who try to understand how the tests works and are not yet `pytest` experts.

Reply:
I don't think we should avoid using `pytest.fixture`. It's one of the core features of `pytest` and I'm not doing anything fancy with it (e.g. auto-applied fixtures), just having it passed to the tests.

Reply:
Seems a little too complicated than necessary for this particular case. Why can't all the methods be combined into one test function running all those tests? You are not doing much with the object here, just testing some parameters.

Reply:
I rewrote the tests to avoid using fixtures. But I kept the tests atomic.